### PR TITLE
Implement-CRUD back-end

### PR DESCRIPTION
**Pull request by VIKTORIAIT**

### CODE REVIEW

File: server/src/controllers/todo.controller.ts [hunk 1]
```diff
@@ -2,13 +2,40 @@ import { Response, Request } from "express";
...
-    async getAllTodo(_: Request, res: Response) {
-        // TODO: Write your implementation here
-        const todos = await this.todoService.findAll();
-        res.send(todos);
-    }
+  constructor(private todoService: TodoService) {}
+
+  async getAllTodo(_: Request, res: Response) {
+    const todos = await this.todoService.findAll();
+    res.send(todos);
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:9]: Handle res send inside tryCatch middleware

File: server/src/controllers/todo.controller.ts [hunk 2]
```diff
@@ -2,13 +2,40 @@ import { Response, Request } from "express";
...
+  }
+
+  async addTodoItem(_: Request, res: Response) {
+    const { body } = _;
+
+    const todos = await this.todoService.addItem(body);
+    res.send(todos);
+  }
+
+  async editTodoItem(_: Request, res: Response) {
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:19]: Add request body and params types

Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:19]: Add isExistMiddleware

File: server/src/controllers/todo.controller.ts [hunk 3]
```diff
@@ -2,13 +2,40 @@ import { Response, Request } from "express";
...
+    const {
+      body: item,
+      params: { id },
+    } = _;
+
+    const todos = await this.todoService.editItem({ id, item });
+    res.send(todos);
+  }
+
+  async deleteTodoItem(_: Request, res: Response) {
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:29]: Types

Comment (CodeGeneration-2020) [reply server/src/controllers/todo.controller.ts:29]: Add isExistMiddleware

File: server/src/controllers/todo.controller.ts [hunk 4]
```diff
@@ -2,13 +2,40 @@ import { Response, Request } from "express";
...
+    const {
+      params: { id },
+    } = _;
+
+    const todos = await this.todoService.deleteItem(id);
+    console.log(todos, "todoooos");
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:35]: Console log

**Review by CodeGeneration-2020 — CHANGES_REQUESTED** [review]

File: server/src/controllers/todo.controller.ts (hunk 1 above)
Comment (VIKTORIAIT) [reply server/src/controllers/todo.controller.ts:9]: Ok, thanks

File: server/src/controllers/todo.controller.ts (hunk 4 above)
Comment (VIKTORIAIT) [reply server/src/controllers/todo.controller.ts:35]: Ok

**Pull request closed without merging**